### PR TITLE
Add working equals & readable toString to NetworkAddressRange

### PR DESCRIPTION
**Pull request by Mahoney**
Also add some defensive validation.

## References

- TODO

## Submitter checklist

- [X] Recommended: Join [WireMock Slack](https://slack.wiremock.org/) to get any help in `#help-contributing` or a project-specific channel like `#wiremock-java`
- [X] The PR request is well described and justified, including the body and the references
- [X] The PR title represents the desired changelog entry
- [X] The repository's code style is followed (see the contributing guide)
- [ ] Test coverage that demonstrates that the change works as expected
- [X] For new features, there's necessary documentation in this pull request or in a subsequent PR to [wiremock.org](https://github.com/wiremock/wiremock.org)

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]

File: src/main/java/com/github/tomakehurst/wiremock/common/NetworkAddressRange.java [hunk 1]
```diff
@@ -32,6 +32,11 @@ public abstract class NetworkAddressRange {
       Pattern.compile(SINGLE_IP.pattern() + "-" + SINGLE_IP.pattern());
 
   public static NetworkAddressRange of(String value) {
+
+    if (value == null || value.isEmpty()) {
+      throw new InvalidInputException(Errors.single(17, value + " is not a valid network address"));
```
Comment (oleg-nenashev) [inline src/main/java/com/github/tomakehurst/wiremock/common/NetworkAddressRange.java:37]: Why 17 by the way?

Comment (Mahoney) [reply src/main/java/com/github/tomakehurst/wiremock/common/NetworkAddressRange.java:37]: Just copying the existing code: https://github.com/wiremock/wiremock/blob/e4dbb74069af90696fc058b809f8fa6783f10e67/src/main/java/com/github/tomakehurst/wiremock/common/NetworkAddressRange.java#L154-L161

Comment (Mahoney) [reply src/main/java/com/github/tomakehurst/wiremock/common/NetworkAddressRange.java:37]: Not sure there's much logic to error codes beyond "uniquely identifies the problem"

Comment (oleg-nenashev) [reply src/main/java/com/github/tomakehurst/wiremock/common/NetworkAddressRange.java:37]: Ok :) 

```suggestion
      throw new InvalidInputException(Errors.single(42, value + " is not a valid network address"));
```

Comment (oleg-nenashev) [reply src/main/java/com/github/tomakehurst/wiremock/common/NetworkAddressRange.java:37]: Nope, NVM. I will merge as is

**Pull request merged**